wgsl/index.bs

```diff
@@ -6835,31 +6835,36 @@ That's not a full user-defined function declaration.
 
 <table class='data'>
   <thead>
-    <tr><th>Precondition<td>Conclusion<td>Notes
+    <tr><th>Precondition<th style="width:45%">Conclusion<td>Description
   </thead>
   <tr algorithm="vector all">
-    <td>|e|: vecN&lt;bool&gt;
-    <td>`all(`|e|`)`: bool
+    <td>|T| is bool
+    <td>`all(`|e|`:` vecN<|T|`> ) -> ` |T|
     <td>Returns true if each component of |e| is true.
     (OpAll)
 
   <tr algorithm="vector any">
-    <td>|e|: vecN&lt;bool&gt;
-    <td>`any(`|e|`)`: bool
+    <td>|T| is bool
+    <td>`any(`|e|`:` vecN<|T|`> ) -> ` |T|
     <td>Returns true if any component of |e| is true.
     (OpAny)
 
   <tr algorithm="scalar select">
-    <td>|T| is a scalar
-    <td>`select(`|f|`: `|T|` , `|t|`: `|T|`, `|cond|`: bool)`: |T|
-    <td>Returns |t| when |cond| is true, and |f| otherwise.
+    <td>|T| is a scalar<br>|I| is bool
+    <td>`select(`|e1|`: ` |T|` , `|e2|`: ` |T|`, `|e3|`: ` |I|` ) -> ` |T|
+    <td>Returns |e1| when |e3| is true, and |e2| otherwise.
     (OpSelect)
 
   <tr algorithm="vector select">
-    <td>|T| is a scalar
-    <td>`select(`|f|`: vecN<`|T|`>, `|t|`: vecN<`|T|`, `|cond|`: vecN<bool>>)`
-    <td>[=Component-wise=] selection. Result component |i| is evaluated as `select(`|f|`[`|i|`], `|t|`[`|i|`], `|cond|`[`|i|`])`.
+    <td>|T| is a scalar<br>|I| is bool
+    <td>`select(`|e1|`: vecN<`|T|`>, `|e2|`: vecN<`|T|`>, `|e3|`: vecN<`|I|`>) -> ` vecN<|T|>
+    <td>[=Component-wise=] selection. Result component |i| is evaluated as `select(`|e1|`[`|i|`], `|e2|`[`|i|`], `|e3|`[`|i|`])`.
     (OpSelect)
+
+  <tr algorithm="vector select with scalar condiction">
```

Comment:
The group did not discuss this one, so it's additional functionality.

Reply:
The WGSL spec used to have this overload, and it looks like it was accidentally removed when this part of the table was reworked to fix an unrelated issue in this PR:
https://github.com/gpuweb/gpuweb/pull/1957/files

We already implement this overload in Tint. It's directly implementably for MSL and HLSL, and a trivial polyfill for SPIR-V.

Reply:
Thanks!   And thanks @sarahM0 for linking back to the original resolution #826.